tests/vmcheck/test-autoupdate.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+#
+# Copyright (C) 2018 Jonathan Lebon
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. ${commondir}/libtest.sh
+. ${commondir}/libvm.sh
+
+set -x
+
+# first, let's make sure the timer is disabled so it doesn't mess up with our
+# tests
+vm_cmd systemctl disable --now rpm-ostreed-automatic.timer
+
+# Really testing this like a user requires a remote ostree server setup.
+# Let's start by setting up the repo.
+REMOTE_OSTREE=/ostree/repo/tmp/vmcheck-remote
+vm_cmd mkdir -p $REMOTE_OSTREE
+vm_cmd ostree init --repo=$REMOTE_OSTREE --mode=archive
+vm_start_httpd ostree_server $REMOTE_OSTREE 8888
+
+# steal a commit from the system repo and make a branch out of it
+lift_commit() {
+  checksum=$1; shift
+  branch=$1; shift
+  vm_cmd ostree pull-local --repo=$REMOTE_OSTREE --disable-fsync \
+    /ostree/repo $checksum
+  vm_cmd ostree --repo=$REMOTE_OSTREE refs $branch --delete
+  vm_cmd ostree --repo=$REMOTE_OSTREE refs $checksum --create=$branch
+}
+
+# use a previously stolen commit to create an update on our vmcheck branch,
+# complete with version string and pkglist metadata
+create_update() {
+  branch=$1; shift
+  vm_cmd ostree commit --repo=$REMOTE_OSTREE -b vmcheck \
+    --tree=ref=$branch --add-metadata-string=version=$branch --fsync=no
+  # avoid libtool wrapper here since we're running on the VM and it would try to
+  # cd to topsrcdir/use gcc; libs are installed anyway
+  vm_cmd /var/roothome/sync/.libs/inject-pkglist $REMOTE_OSTREE vmcheck
+}
+
+# (delete ref but don't prune for easier debugging)
+vm_cmd ostree refs --repo=$REMOTE_OSTREE vmcheck --delete
+
+# now let's build some pkgs that we'll jury-rig into a base update
+# this whole block can be commented out for a speed-up when iterating locally
+vm_build_rpm base-pkg-foo version 1.4 release 7
+vm_build_rpm base-pkg-bar
+vm_build_rpm base-pkg-baz version 1.1 release 1
+vm_rpmostree install base-pkg-{foo,bar,baz}
+lift_commit $(vm_get_pending_csum) v1
+vm_rpmostree cleanup -p
+rm -rf $test_tmpdir/yumrepo
+vm_build_rpm base-pkg-foo version 1.4 release 8 # upgraded
+vm_build_rpm base-pkg-bar version 0.9 release 3 # downgraded
+vm_build_rpm base-pkg-boo version 3.7 release 2.11 # added
+vm_rpmostree install base-pkg-{foo,bar,boo}
+lift_commit $(vm_get_pending_csum) v2
```

Comment:
Interesting the way you're using client-side `install` to generate server-side commits.  I paused at this; probably just worth a comment?